name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/ibflex/IBFlexStatementExtractor.java

```diff
@@ -338,103 +290,139 @@ else if (this.ibAccountCurrency.equals(s.getCurrencyCode()))
-            {
-                transaction.setType(PortfolioTransaction.Type.BUY);
-            }
-            else if (element.getAttribute("buySell").equals("SELL"))
-            {
-                transaction.setType(PortfolioTransaction.Type.SELL);
-            }
-            else
+            // Set transaction type
+            String buySell = element.getAttribute("buySell");
+            switch (buySell)
             {
-                throw new IllegalArgumentException();
+                case "BUY":
+                    transaction.setType(PortfolioTransaction.Type.BUY);
+                    break;
+                case "BUY (Ca.)":
+                    transaction.setNote(Messages.MsgErrorOrderCancellationUnsupported);
+                    transaction.setType(PortfolioTransaction.Type.SELL);
+                    break;
+                case "SELL":
+                    transaction.setType(PortfolioTransaction.Type.SELL);
+                    break;
+                default:
+                    throw new IllegalArgumentException();
             }
 
             // Sometimes IB-FlexStatement doesn't include "tradeDate" - in this
             // case tradeDate will be replaced by "000000".
             // New format is stored in dateTime, take care for double imports).
             if (element.hasAttribute("dateTime"))
             {
-                transaction.setDate(convertDate(element.getAttribute("dateTime").substring(0, 8),
+                transaction.setDate(ExtractorUtils.asDate(element.getAttribute("dateTime").substring(0, 8),
                                 element.getAttribute("dateTime").substring(9, 15)));
             }
             else
             {
                 if (element.hasAttribute("tradeTime"))
-                {
-                    transaction.setDate(
-                                    convertDate(element.getAttribute("tradeDate"), element.getAttribute("tradeTime")));
-                }
-                else
-                {
-                    transaction.setDate(convertDate(element.getAttribute("tradeDate"), "000000"));
-                }
+                    transaction.setDate(ExtractorUtils.asDate(element.getAttribute("tradeDate"),
+                                    element.getAttribute("tradeTime")));
             }
 
-            // transaction currency
-            String currency = asCurrencyUnit(element.getAttribute("currency"));
+            // Set transaction currency
+            String currency = asCurrencyCode(element.getAttribute("currency"));
 
-            // Set the Amount which is "netCash"
-            Double amount = Math.abs(Double.parseDouble(element.getAttribute("netCash")));
-            setAmount(element, transaction.getPortfolioTransaction(), amount, currency);
-            setAmount(element, transaction.getAccountTransaction(), amount, currency, false);
+            // Set amount which is "netCash" or "cost"
```

Comment:
Here's where the problem is - the code doesn't correspond to the comment, it really should be "netCash", then "cost" (that's assuming there can be a case when "cost" can go without "netCash", after all the old code used just netCash).

Reply:
Hello @pfalcon 
Thanks 👍🏻 
I have tested all the files I found in the PP forum. Can you provide me with a full XML file with some examples?
Also the results for each transaction?
That would be great, then I could check this and adjust the importer if necessary in case of errors.